File: Core/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public Order()
        {
        }

        public Order(IReadOnlyList<OrderItem> orderItems, string buyerEmail, Address address, DeliveryMethod deliveryMethod, decimal subtotal)
        {
            BuyerEmail = buyerEmail ?? throw new ArgumentNullException(nameof(buyerEmail));
            Address = address ?? throw new ArgumentNullException(nameof(address));
            DeliveryMethod = deliveryMethod ?? throw new ArgumentNullException(nameof(deliveryMethod));
            OrderItems = orderItems ?? throw new ArgumentNullException(nameof(orderItems));
            Subtotal = subtotal;
        }

        public string BuyerEmail { get; set; }
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;
        public Address Address { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }
        public decimal GetTotal() 
        {
            return Subtotal + DeliveryMethod.Price;
        }
    }
}

File: Client/src/app/shared/_models/order.ts
import { Address } from './address';

export interface OrderToCreate {
  basketId: string;
  deliveryMethod: number;
  address: Address;
}

export interface Order {
  id: number;
  buyerEmail: string;
  orderDate: string;
  shipToAddress: Address;
  deliveryMethod: string;
  shippingPrice: number;
  orderItems: OrderItem[];
  subtotal: number;
  total: number;
  status: string;
}

export interface OrderItem {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}
